python-checks/src/main/java/org/sonar/python/checks/InstanceAndClassMethodsAtLeastOnePositionalCheck.java

```diff
@@ -39,6 +40,16 @@
 
   private static final List<String> KNOWN_CLASS_METHODS = Arrays.asList("__new__", "__init_subclass__");
 
+  private static boolean isUsageInClassBody(Usage usage, ClassDef classDef) {
+    if (usage.kind() == Usage.Kind.FUNC_DECLARATION) {
+      return false;
+    }
+
+    Tree tree = usage.tree();
+    return classDef.equals(TreeUtils.firstAncestorOfKind(tree, Tree.Kind.CLASSDEF))
```

Comment:
I think you can refactor this code:
```java
 return classDef.equals(TreeUtils.firstAncestorOfKind(tree, Tree.Kind.CLASSDEF, Tree.Kind.FUNCDEF))
```

Reply:
Fixed in 66d6ff506e3872d642cfd6b1be922a8ac59be928